src/App.tsx

```diff
@@ -1,57 +1,53 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash/debounce';
 import { peopleFromServer } from './data/people';
+import { AppContext } from './AppContext';
+import { People } from './components/People';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [people, setPeople] = useState(peopleFromServer);
+  const [personName, setPersonName] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState('No Person is selected');
+
+  const [applyQuery, setApplyQuery] = useState('');
+
+  const appliedQuery = useCallback(
+    (value) => debounce(setApplyQuery, 1000)(value),
+    [],
+  );
+
+  const filterPeople = useMemo(() => {
+    return peopleFromServer.filter(
+      person => person.name.toLowerCase().includes(applyQuery.toLowerCase()),
+    );
+  }, [people, applyQuery]);
```

Comment:
![image](https://github.com/mate-academy/react_autocomplete/assets/70009205/a798d381-27a1-46d5-8db7-14589df8a3e7)
```suggestion
  }, [applyQuery]);
```